astropy/cosmology/core.py

```diff
@@ -1374,23 +1375,27 @@ def luminosity_distance(self, z):
 
     def angular_diameter_distance_z1z2(self, z1, z2):
         """ Angular diameter distance between objects at 2 redshifts.
-        Useful for gravitational lensing.
+        Useful for gravitational lensing, for example computing the angular diameter
+        distance between a lensed galaxy and the foreground lens.
 
         Parameters
         ----------
         z1, z2 : array_like, shape (N,)
-          Input redshifts. z2 must be large than z1.
+          Input redshifts. For most practical applications such as gravitational lensing,
+          z2 should be larger than z1.  The method will work for z2<z1; however, it may
```

Comment:
```suggestion
          z2 should be larger than z1.  The method will work for z2<z1; however, this will
```
might as well be unambiguous here – at least I am fairly sure it will _always_ be negative in this case, some comments on possible very exotic cosmologies in the original thread nonwithstanding.
As a matter of fact all existing `scalar_inv_efuncs` return a `sqrt`, so `_integral_comoving_distance_z1z2` integrating in negative direction will definitely be negative.